client/src/components/posts/postReactions/Comments/CmntFuncionality.tsx

```diff
@@ -0,0 +1,184 @@
+    setComments: (comments: Comment[]) => void;
+}
+
+
+export const CmntFunctionality: React.FC<CmntFunctionalityProps> = ({
+    comments,
+    setComments,
+}) => {
+
+    const handleSave = (commentId: number, newText: string) => {
+
+        setComments(
+            comments.map((comment) => {
+                if (comment.id == commentId && comment.text != '') {
+
+                    return { ...comment, text: newText, editable: false };
+                }
+                return comment;
+            })
+        )
+    }
+    const handleEdit = (commentId: number) => {
+        setComments(
+            comments.map((comment) => {
+                if (comment.id === commentId) {
+                    return { ...comment, editable: true };
+                }
+                return comment;
+            })
+        )
+    }
+    const handleDelete = (commentId: number) => {
+        setComments(comments.filter((comment) => comment.id !== commentId));
+    }
+
+
+
+    return (
+        <>
+            {
+                comments.length > 0 ? (
+                    <Grid className='test' sx={{
+                        paddingTop: 1,
+                        borderRadius: 5,
+                        paddingBottom: 1,
+                        maxWidth: 500,
+                        maxHeight: 3000,
+                        backgroundColor: 'white',
+                        boxShadow: 'none',
+
+                    }} >
+                        {
+                            comments.map((comment) => (
+                                <Grid key={comment.id}>
+                                    {
+                                        comment.editable ? (
+                                            <>
+                                                <Divider variant="middle" sx={{ marginTop: 2 }} />
+                                                <ListItem alignItems="flex-start">
+                                                    <ListItemAvatar>
+                                                        <Avatar sx={{ marginLeft: 2, backgroundColor: 'green' }}>R</Avatar>
+                                                    </ListItemAvatar>
+                                                    <ListItemText
+                                                        sx={{ marginLeft: 2 }}
+                                                        primary="username"
+                                                        secondary={
+                                                            <React.Fragment>
+                                                                <TextareaAutosize
+                                                                    spellCheck="false"
+                                                                    className='editTextArea'
+                                                                    value={comment.text}
+                                                                    onChange={(e) =>
```

Comment:
Please try to use more expensive variable names such as:  "event". It is more understandable and also you  avoid signle character variables.